crates/aptos/src/move_tool/mod.rs

```diff
@@ -1500,18 +1504,17 @@ impl CliCommand<TransactionSummary> for DeployObjectCode {
                     MAX_PUBLISH_PACKAGE_SIZE,
                 ));
             }
-            self.txn_options
-                .submit_transaction(payload)
-                .await
-                .map(TransactionSummary::from)
+            dispatch_transaction(payload, &self.txn_options).await
         };
 
-        if result.is_ok() {
+        if let Ok(tx_summary) = &mut result {
             println!(
                 "Code was successfully deployed to object address {}",
                 object_address
             );
+            tx_summary.object_address = Some(object_address);
```

Comment:
Now I see, `object_address` is where code is deployed? Probably it is better to change the name to reflect this.